src/models/semantic.py

```diff
@@ -817,22 +836,30 @@ def on_test_epoch_end(self) -> None:
             self.test_cm.reset()
             return
 
+        if self.trainer.num_devices > 1:
+            epoch_cm = torch.sum(self.all_gather(self.test_cm.confmat), dim=0)
+        else:
+            epoch_cm = self.test_cm
+        # Reset metrics accumulated over the last epoch
+        self.test_cm.reset()
+
+        epoch_cm = ConfusionMatrix(self.num_classes).from_confusion_matrix(epoch_cm)
```

Comment:
This is multi-device-specific. Please move in the if statement after you gather the confmats from each device